### PR TITLE
Updating heat template for mitaka

### PR DESCRIPTION
**Pull request by sayalilunkad**

### CODE REVIEW

**Comment by jgrassler** [comment]
:+1:

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 1]
```diff
@@ -32,6 +29,16 @@
 #deferred_auth_method = trusts
 deferred_auth_method = trusts
 
+# Allow reauthentication on token expiry, such that long-running tasks may
+# complete. Note this defeats the expiry of any provided user tokens. (string
+# value)
+# Allowed values: '', trusts
+#reauthentication_auth_method =
```
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Do we want to have this set to ```trusts``` ?

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 2]
```diff
@@ -621,13 +711,77 @@ endpoint_type=internalURL
 insecure = <%= @keystone_settings['insecure'] %>
 <% end -%>
 
+# Unversioned keystone url in format like http://0.0.0.0:5000. (string value)
+#auth_uri =
```
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: I think this should be ```<%= @keystone_settings['public_auth_url'] %>```

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 3]
```diff
@@ -621,13 +711,77 @@ endpoint_type=internalURL
...
+#ca_file = <None>
+
+# Optional PEM-formatted certificate chain file. (string value)
+#cert_file = <None>
+
+# Optional PEM-formatted file that contains the private key. (string value)
+#key_file = <None>
+
+# If set, then the server's certificate will not be verified. (boolean value)
+#insecure = <None>
```
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Manila has already SSL support so we need to set this

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 4]
```diff
@@ -621,13 +711,77 @@ endpoint_type=internalURL
...
+
+[clients_manila]
+
+#
+# From heat.common.config
+#
+
+# Type of endpoint in Identity service catalog to use for communication with
+# the OpenStack service. (string value)
+#endpoint_type = <None>
```
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Should be ```endpoint_type=internalURL``` I think

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 5]
```diff
@@ -472,6 +523,29 @@ insecure = <%= @keystone_settings['insecure'] %>
...
 
+[clients_barbican]
+
+#
+# From heat.common.config
+#
+
+# Type of endpoint in Identity service catalog to use for communication with
+# the OpenStack service. (string value)
+#endpoint_type = <None>
```
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Should be ```endpoint_type=internalURL``` I think

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 6]
```diff
@@ -621,13 +711,77 @@ endpoint_type=internalURL
...
 #
 # From heat.common.config
 #
 
-# Unversioned keystone url in format like http://0.0.0.0:5000. (string value)
-#auth_uri =
-auth_uri = <%= @keystone_settings['protocol'] %>://<%= @keystone_settings['public_url_host'] %>:<%= @keystone_settings['service_port'] %>
+# Type of endpoint in Identity service catalog to use for communication with
+# the OpenStack service. (string value)
+#endpoint_type = <None>
```
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Should be ```endpoint_type=internalURL``` I think

File: chef/cookbooks/heat/templates/default/heat.conf.erb (hunk 1 above)
Comment (jgrassler) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Yes, this is the way to go. The alternative (the old way) is quite insane: See the section "What? Passwords? Don't we pass tokens?" in http://hardysteven.blogspot.de/2014/04/heat-auth-model-updates-part-1-trusts.html for a description.

tl;dr: If this is not set to `trusts` the user will have to pass their password to Heat which will store it in the database so Heat can use it for long-running tasks...

File: chef/cookbooks/heat/templates/default/heat.conf.erb [hunk 7]
```diff
@@ -1596,19 +1968,84 @@ rabbit_virtual_host=<%= @rabbit_settings[:vhost] %>
...
+# SSLv23. SSLv2, SSLv3, TLSv1_1, and TLSv1_2 may be available on some
+# distributions. (string value)
+#version = <None>
+
+# Sets the list of available ciphers. value should be a string in the OpenSSL
+# cipher list format. (string value)
+#ciphers = <None>
+
+
+[trustee]
```
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Don't we want to setup the trusts delegation? @jgrassler  ?

File: chef/cookbooks/heat/templates/default/heat.conf.erb (hunk 1 above)
Comment (toabctl) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: @sayalilunkad can you set this in another commit or create a follow-up card?

File: chef/cookbooks/heat/templates/default/heat.conf.erb (hunk 7 above)
Comment (jgrassler) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: If I remember correctly, the (sensible) default is to delegate trusts to the Heat service user, but let me double-check...

Comment (jgrassler) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Interesting. This has been deprecated in favour of explicitly configuring a trustee user's credentials in the `[trustee]` section: https://github.com/openstack/heat/commit/67f12e5e24eb9ec06e609f4399008b9d704128a1

I suppose we should configure this explicitly then, just in case hard deprecation throws a spanner in the works at some later stage (I don't think it will happen in Newton - `master` still contains the code issuing the deprecation warning). In the scope of this pull request just mirroring the credentials from the `[keystone_authtoken]` section should be enough, though.

File: chef/cookbooks/heat/templates/default/heat.conf.erb (hunk 1 above)
Comment (sayalilunkad) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Done.

File: chef/cookbooks/heat/templates/default/heat.conf.erb (hunk 2 above)
Comment (sayalilunkad) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Done.

File: chef/cookbooks/heat/templates/default/heat.conf.erb (hunk 7 above)
Comment (sayalilunkad) [inline chef/cookbooks/heat/templates/default/heat.conf.erb]: Done.

**Comment by toabctl** [comment]
+1

**Comment by jgrassler** [comment]
+1
